managed/services/agents/roster.go

```diff
@@ -35,34 +39,62 @@ const (
 type roster struct {
 	l *logrus.Entry
 
+	db *reform.DB
 	rw sync.RWMutex
 	m  map[string][]string
 }
 
-func newRoster() *roster {
+func newRoster(db *reform.DB) *roster {
 	return &roster{
-		l: logrus.WithField("component", "roster"),
-		m: make(map[string][]string),
+		db: db,
+		l:  logrus.WithField("component", "roster"),
+		m:  make(map[string][]string),
 	}
 }
 
-func (r *roster) add(pmmAgentID string, group agentGroup, agentIDs []string) (groupID string) { //nolint:nonamedreturns
+func (r *roster) add(pmmAgentID string, group agentGroup, exporters map[*models.Node]*models.Agent) string {
 	r.rw.Lock()
 	defer r.rw.Unlock()
 
-	groupID = pmmAgentID + "/" + string(group)
-	r.m[groupID] = agentIDs
-	r.l.Debugf("add: %s = %v", groupID, agentIDs)
-	return
+	groupID := pmmAgentID + "/" + string(group)
+	exporterIDs := make([]string, 0, len(exporters))
+	for _, exporter := range exporters {
+		exporterIDs = append(exporterIDs, exporter.AgentID)
+	}
+
+	sort.Strings(exporterIDs)
+
+	r.m[groupID] = exporterIDs
+	r.l.Debugf("add: %s = %v", groupID, exporterIDs)
+	return groupID
 }
 
-func (r *roster) get(groupID string) (agentIDs []string) { //nolint:nonamedreturns
+func (r *roster) get(groupID string) (string, []string, error) {
```

Comment:
i know the linter frowns at named returns - maybe we can add method comments to describe what each return parameter is/does?

Reply:
Sure. Done!